### ISSUE TITLE
Custom properties in test model items

### ISSUE THREAD

**Issue by jonherrmann** [etf-validator/governance#25]
**ETF Improvement Proposal (EIP)**

# Background and Motivation:
The specification of meta-information are very limited in an ETS. Currently
it is not possible to enrich test model items with additional machine-readable
information in a structured way. These information could be used by:

- custom reports that show additional meta-information of the test model item
- other (external) clients that access and parse the ETS

Example: store an alternative name for an test item. The description field
could be used to store additional information but the format would be
unspecified and could not easily processed by a machine.

# Proposed change
Support adding a sequence of custom properties to a test model item.

The properties can have a name and value. Both are character strings.
Exampe:

```xml
<Properties>
    <property name="Alternative Name">ExecutableTestsuites.X</property>
    <property name="Another custom property name">another custom proeprty value</property>
</Properties>
```

The properties would be added to all test items, the Executable Testsuites,
TestModules, TestCases, TestSteps and TestAssertions. The Properties Element
would be optional.

```xml
<ExecutableTestSuite >
    ...
    <label>Conformance Class: ETS X</label>
    <description>Description of the test suite</description>
    ...
    <Properties>
        <property name="Alternative Name">ExecutableTestsuites.X</property>
        <property name="Another custom property name">another custom proeprty value</property>
    </Properties>        
    ...
</ExecutableTestSuite>
```

# Alternatives
This information could also be stored separately from the test model items in
another structure. The implementation effort would be higher and the processing
of this information would be more time consuming because the information would
have to be retrieve from different sources.

# Funding
Full funding available.

# Additional information
- This EIP is backwards compatible with existing ETSs
- The change would affect the ETF core module and the ETF data storage

**Comment by jonherrmann** [comment]
Scheduled for ETF-Webapp release 2.1.0.

Component versions:
- etf-core 1.2.0
- etf-bsxds-2.1.0

**Comment by jonherrmann** [comment]
Implemented in [Version 2.1.0](https://github.com/etf-validator/etf-webapp/releases/tag/2.1.0)
